requestparsing/test_harness.go

```diff
@@ -0,0 +1,120 @@
+package requestparsing
+
+import (
+	"context"
+	"errors"
+	"io"
+	"net"
+	"net/http"
+	"sync"
+)
+
+type fakeHandler struct {
```

Comment:
Renamed in 76e7795.

Why should this struct be public (exported)? We will only use it in this package

Reply:
The setup seems to be good enough to be useful for more than just the "requestparsing" package, therefore I thought it might be a good idea to make it public and put into a `testing` package.

Reply:
The main reason is: I imagine this package being quite useful. Keeping it inside a "requestparsing" package makes it visible only to it, therefore no tests in other packages can be made.

Reply:
Renamed package in [527b6aa](url)